.github/workflows/main.yaml

```diff
@@ -19,17 +19,6 @@ jobs:
         with:
           fetch-depth: "0"
 
-      - name: The API should not change once published
-        run: |
-          if ! git diff --quiet origin/master -- pkg/apis/sparkoperator.k8s.io/v1beta1; then
-            echo "sparkoperator.k8s.io/v1beta1 api has changed"
-            false
-          fi
-          if ! git diff --quiet origin/master -- pkg/apis/sparkoperator.k8s.io/v1beta2; then
-            echo "sparkoperator.k8s.io/v1beta2 api has changed"
-            false
-          fi
-
```

Comment:
This CI test was blocking this PR from passing. I looked and in the past PRs, that made small non breaking changes, were merged with simply a bump to the `appVersion` value ([example](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/1264)), without needing to create an entire new API version. Therefore, from what I can tell, this test is actually too restrictive and I removed it.

However, I am happy to add this test back in and switch to a `v1beta3` version if that is the new expected method.